data-prepper-api/src/main/java/org/opensearch/dataprepper/metrics/MetricNames.java

```diff
@@ -57,6 +57,11 @@ private MetricNames() {}
      */
     public static final String RECORDS_PROCESSED = "recordsProcessed";
 
+    /**
+     * Metric representing the number of records failed to be written into the buffer.
+     */
+    public static final String RECORDS_OVERFLOW = "recordsOverflow";
```

Comment:
Perhaps `recordsWriteFailed` would be clearer? We are not overflowing the buffer, but instead failed to write to the buffer.